### PR TITLE
Disallow sliding windows with combiner fanout to prevent data loss

### PR DESCRIPTION
**Pull request by kennknowles**
A temporary quick fix for #20528 until we can make it work entirely.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/transforms/core.py [hunk 1]
```diff
@@ -2746,6 +2747,11 @@ def expand(self, pcoll):
     combine_fn = self._combine_fn
     fanout_fn = self._fanout_fn
 
+    if self.isinstance(pcoll.windowfn, SlidingWindows):
+      raise ValueError(
+          'CombinePerKey.with_hot_key_fanout does not yet work properly with SlidingWindows'
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/core.py:2752]: Thanks.

please reference: https://github.com/apache/beam/issues/20528.

I recommend you import this change internally first before cutting the release candidate.

**Review by tvalentyn — APPROVED** [review]
Consider adding  test_combining_with_sliding_windows_and_fanout from https://github.com/apache/beam/commit/42688356bd27cd1db0e1b03212249d7f424d276c and verifying that an exception was raised.

File: sdks/python/apache_beam/transforms/core.py [hunk 2]
```diff
@@ -2746,6 +2747,11 @@ def expand(self, pcoll):
     combine_fn = self._combine_fn
     fanout_fn = self._fanout_fn
 
+    if self.isinstance(pcoll.windowfn, SlidingWindows):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/transforms/core.py:2750]: actually, this statement wouldn't work.

Comment (tvalentyn) [reply sdks/python/apache_beam/transforms/core.py:2750]: I'll push a fix.

Comment (kennknowles) [reply sdks/python/apache_beam/transforms/core.py]: Oh, derp. I didn't notice I had put `self.` in there. Yeesh

**Review by tvalentyn — APPROVED** [review]

**Pull request merged**